### PR TITLE
fix(ingest): lookml - add support for includes, extends, view_name i…

### PR DESCRIPTION
**Pull request by shirshanka**
…n explore

Fixes a few issues with discovering views when navigating explores in the lookml source. 
- view_name is used for the name of the view when present
- explores that live in included files (and their includes) are now included in the traversal
- extends directives in explores are handled

## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/looker/looker_common.py [hunk 1]
```diff
@@ -536,11 +537,17 @@ def from_dict(
         resolved_includes: List[ProjectInclude],
         looker_viewfile_loader: "LookerViewFileLoader",
         reporter: "LookMLSourceReport",
+        model_explores_map: Dict[str, dict],
     ) -> "LookerExplore":
-        view_names = set()
+
+        view_names: Set[str] = set()
```
Comment (gabe-lyons) [inline metadata-ingestion/src/datahub/ingestion/source/looker/looker_common.py:543]: nice!

File: metadata-ingestion/src/datahub/ingestion/source/looker/looker_common.py [hunk 2]
```diff
@@ -536,11 +537,17 @@ def from_dict(
...
         joins = None
-        # always add the explore's name or the name from the from clause as the view on which this explore is built
-        view_names.add(dict.get("from", dict.get("name")))
+        assert "name" in dict, "Explore doesn't have a name field, this isn't allowed"
```
Comment (gabe-lyons) [inline metadata-ingestion/src/datahub/ingestion/source/looker/looker_common.py:545]: asking to learn: if someone made a lookml like this, why isn't it allowed?

Comment (gabe-lyons) [reply metadata-ingestion/src/datahub/ingestion/source/looker/looker_common.py:545]: would this lookml also not be accepted by looker or is a datahub specific requirement?

Comment (shirshanka) [reply metadata-ingestion/src/datahub/ingestion/source/looker/looker_common.py:545]: the lkml parser will populate the `name` field in the dict based on the explore definition, that is the contract.
So, it's not about having a name field inside the explore definition... more like parsing the explore will populate the name field. 

e.g.
```
explore: events {
  ... explore def
}
```
will result in a dictionary with the `name` key set to `events`

File: metadata-ingestion/src/datahub/ingestion/source/looker/looker_common.py [hunk 3]
```diff
@@ -559,23 +566,47 @@ def from_dict(
...
 
-        upstream_views = []
-        for view_name in view_names:
-            info = _find_view_from_resolved_includes(
-                None,
-                resolved_includes,
-                looker_viewfile_loader,
-                view_name,
-                reporter,
+        upstream_views: List[ProjectInclude] = []
```
Comment (gabe-lyons) [inline metadata-ingestion/src/datahub/ingestion/source/looker/looker_common.py:569]: woo!

File: metadata-ingestion/src/datahub/ingestion/source/looker/looker_common.py [hunk 4]
```diff
@@ -559,23 +566,47 @@ def from_dict(
...
             )
-            if not info:
-                logger.warning(
-                    f'Could not resolve view {view_name} for explore {dict["name"]} in model {model_name}'
-                )
-            else:
-                upstream_views.append(
-                    ProjectInclude(project=info[0].project, include=view_name)
+        )
+        if extends:
```
Comment (gabe-lyons) [inline metadata-ingestion/src/datahub/ingestion/source/looker/looker_common.py:576]: nice!

File: metadata-ingestion/src/datahub/ingestion/source/looker/lookml_source.py [hunk 5]
```diff
@@ -363,6 +367,24 @@ def from_looker_dict(
...
+            try:
+                with open(included_file, "r") as file:
+                    parsed = lkml.load(file)
+                    included_explores = parsed.get("explores", [])
+                    explores.extend(included_explores)
+            except Exception as e:
+                reporter.report_warning(
+                    path, f"Failed to load {included_file} due to {e}"
+                )
+                # continue in this case, as it might be better to load and resolve whatever we can
```
Comment (gabe-lyons) [inline metadata-ingestion/src/datahub/ingestion/source/looker/lookml_source.py:385]: yes, agreed!

File: metadata-ingestion/tests/integration/lookml/lkml_samples/data.model.lkml [hunk 6]
```diff
@@ -5,6 +5,7 @@ include: "bar.view.lkml"
 include: "nested/*"
 include: "liquid.view.lkml"
 include: "ability.view.lkml"
+include: "dataset_owners.explore.lkml"
```
Comment (gabe-lyons) [inline metadata-ingestion/tests/integration/lookml/lkml_samples/data.model.lkml:8]: thanks for adding a test

**Review by gabe-lyons — APPROVED** [review]

File: metadata-ingestion/src/datahub/ingestion/source/looker/looker_common.py (hunk 2 above)
Comment (gabe-lyons) [reply metadata-ingestion/src/datahub/ingestion/source/looker/looker_common.py:545]: ah I see, so we'd be looking at a malformed explore then if this wasn't parsed

**Pull request merged**